File: python/sincfun.py
#!/usr/bin/env python3
# File: sincfun.py
# Name: D.Saravanan
# Date: 24/05/2021
# Script to plot normalized sinc function

import math
import matplotlib.pyplot as plt
import matplotlib
matplotlib.use('Agg')

def sinc(value: float) -> float:
    """ sinc function """
    return math.sin(math.pi * value)/(math.pi * value)

xval, yval = [], []
for nval in range(1,101):
    xval.append(nval)
    yval.append(sinc(nval))

plt.plot(xval, yval)
plt.savefig('sincfun.png')

File: julia/sincfun.jl
#!/usr/bin/env julia
# File: sincfun.jl
# Name: D.Saravanan
# Date: 19/08/2021

""" Program to plot normalized sinc function """

using Plots

#@time function sinc(x::Float64)::Float64
#    return sin(pi*x)/(pi*x)
#end

@time x = -2*pi:0.001:2*pi
@time y = [sinc(n) for n in x]

plot(x, y, title = "sinc function")
savefig("sincfun.png")
